conans/test/functional/subsystems_build_test.py

```diff
@@ -88,6 +89,64 @@ def test_mingw64(self):
         assert "__CYGWIN__" not in client.out
         assert "__MSYS__" not in client.out
 
+    @pytest.mark.tool("msys2")
+    @pytest.mark.tool("mingw64")
+    def test_mingw64_recipe(self):
+        """
+        A recipe with self.run_bash=True and msys2 configured, using mingw to build stuff with make
+        from the subsystem
+        """
+        client = TestClient()
+        makefile = gen_makefile(apps=["app"])
+        main_cpp = gen_function_cpp(name="main")
+        conanfile = textwrap.dedent("""
+        import os
+        from conan import ConanFile
+        from conan.tools.gnu import Autotools
+        from conan.tools.layout import basic_layout
+        from conan.tools.files import copy
+        class HelloConan(ConanFile):
+            exports_sources = "*.cpp", "Makefile"
+            generators = "AutotoolsToolchain"
+            win_bash = True
+
+            def build(self):
+                self.output.warning(self.build_folder)
+                auto = Autotools(self)
+                auto.make()
+
+            def package(self):
+                copy(self, "app*", self.build_folder, os.path.join(self.package_folder, "bin"))
+
+        """)
+        test_conanfile = textwrap.dedent("""
+                import os
+                from conan import ConanFile
+                class TestConan(ConanFile):
+
+                    def requirements(self):
+                        self.tool_requires(self.tested_reference_str)
+
+                    def test(self):
+                        self.run("app")
+                """)
+        profile = textwrap.dedent("""
+        include(default)
+        [conf]
+        tools.microsoft.bash:subsystem=msys2
+        tools.microsoft.bash:path=bash.exe
+        tools.gnu:make_program=mingw32-make.exe
```

Comment:
These maybe reasonable defaults (once msys2 and gcc are defined)?
I mean, these will probably take these values 95% of the time, so consider simplifying and defaulting to them (conditioned to msys2, gcc in Windows)

Reply:
Actually, it seems this is already the default for mingw32-make?
```python
make_program = self._conanfile.conf.get("tools.gnu:make_program",
                                                default="mingw32-make" if self._use_win_mingw() else "make")
```

I'd say we do the same default for ``bash.exe``

Reply:
yes, the `mingw32-make` has already defaulted. About the `bash.exe`, it might be a good default, I would change it, but it might be possible (also in the test specifying `tools.microsoft.bash:path=bash.exe`) that it finds another `bash.exe` from `git` or another subsystem, it would be quite confusing but the user can change it anyway...